enhancements/cnf-tests-split.md

```diff
@@ -0,0 +1,204 @@
+
+
+This change will also allow us to pin the external repositories to the latest branch, ensuring that we always test the latest tests without the need for manual commit hash updates. We will include an option to set an environment variable that specifies the commit ref for each repo.
+
+For detailed information about the changes in the cnf-tests container, please refer to the section titled [Rewrite the cnf-tests container](#rewrite-the-cnf-tests-container).
+
+### Goals
+
+1. Stop vendoring the external test suites and run the tests directly from the projects.
+2. Maintain the same API for running the tests.
+3. Preserve the same testing report output.
+4. Refactor the cnf-tests container to contain only the latency-tests.
+
+### Non-Goals
+
+1. Rename existing tests to a certain standard.
+2. Change the scope of the tests being run.
+
+## Proposal
+### User Stories
+#### Story 1
+
+As a developer, I want to run the cnf-tests suites so that I can validate my CNF feature works properly.
+
+#### Story 2
+As an Openshift administrator, I want to run the latency tests so that
+I can benchmark different CNF features.
+
+### Workflow Description
+
+This change will not affect the current API, including parameters/environment variables used to invoke the cnf-tests: `TEST_SUITES`, `FEATURES`, `FOCUS_TESTS`, `SKIP_TESTS`, `GINKGO_PARAMS`, `FAIL_FAST`, and `TESTS_REPORTS_PATH`. 
+However, with this proposal, instead of running test binaries that combine all the CNF components,  we will introduce an **optional** parameter named `CNF_COMPONENTS` to control what external test suites to run.  
+By default, the test will run all of the components it ran prior to this change.
+
+To explain this new variable further, read the following example:
+Running the `e2esuite` test suite, with an empty `CNF_COMPONENTS` and `FEATURES=metallb`, will result in 
+running ginkgo's test in all of the components: sriov, metallb, nto-performance, nto-latency, ptp and integration, adding the flag `-ginkgo.focus=metallb`.
+So in other words, the functionallity of the `FEATURES` var stays the same.
+
+When running the `e2esuite` test suite with `CNF_COMPONENTS=metallb` it will invoke only the metallb component
+of the e2esuite, i.e.: `cnf-tests/submodules/metallb-operator/test/e2e/functional/tests`
+and for the validation suite, it will run: `cnf-tests/submodules/metallb-operator/test/e2e/validation/tests`
+
+The mapping of suite + component to the external test suite path is done by:
+```bash
+declare -A TESTS_PATHS=\
+(["configsuite nto"]="cnf-tests/submodules/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/0_config"\
+ ["validationsuite cluster"]="cnf-tests/testsuites/validationsuite"
+ ["validationsuite metallb"]="cnf-tests/submodules/metallb-operator/test/e2e/validation"
+ ["cnftests integration"]="cnf-tests/testsuites/e2esuite"\
+ ["cnftests metallb"]="cnf-tests/submodules/metallb-operator/test/e2e/functional"\
```

Comment:
Can we have paths like

cnf-tests/external/validation/metallb-operator
cnf-tests/external/validation/sriov

and 

cnf-tests/external/e2e/metallb-operator
cnf-tests/external/e2e/sriov

or something like that to make the navigability easier?

Reply:
This map needs to point to the full path of the test suite in order to invoke it later in the script.
Do you suggest creating soft links to those paths while still cloning to the submodule dir?

Reply:
is the name of the submodule mandatory to be "submodule" ? do we really care if a test suite that we find in a local path comes from a submodule or not?

Reply:
can you please elaborate on how will this work when a test suite in a local path comes not from a submodule?

Reply:
Let's remove the external part then.

cnf-tests/validation/metallb-operator
cnf-tests/validation/sriov
cnf-tests/validation/integration

and

cnf-tests/e2e/metallb-operator
cnf-tests/e2e/sriov
cnf-tests/e2e/integration

Is it possible? And in general, can we think about a dir structure where it's clear what's where without needing the map?

Reply:
ah now I understand. Well, it's not optimal and it makes the navigability of hte project a bit harder. On the other hand I am not sure reshuffling the folders of a submodule makes sense, nor having symlinks.